controllers/operandrequest/reconcile_operand.go

```diff
@@ -200,69 +200,57 @@ func (r *Reconciler) reconcileCRwithConfig(ctx context.Context, service *operato
 }
 
 // reconcileCRwithRequest merge and create custom resource base on OperandRequest and CSV alm-examples
-func (r *Reconciler) reconcileCRwithRequest(ctx context.Context, requestInstance *operatorv1alpha1.OperandRequest, operand operatorv1alpha1.Operand, requestKey types.NamespacedName, csv *olmv1alpha1.ClusterServiceVersion) error {
-	almExamples := csv.GetAnnotations()["alm-examples"]
+func (r *Reconciler) reconcileCRwithRequest(ctx context.Context, requestInstance *operatorv1alpha1.OperandRequest, operand operatorv1alpha1.Operand, requestKey types.NamespacedName) error {
+	merr := &util.MultiErr{}
 
-	// Convert CR template string to slice
-	var almExampleList []interface{}
-	err := json.Unmarshal([]byte(almExamples), &almExampleList)
-	if err != nil {
-		return errors.Wrapf(err, "failed to convert alm-examples in the Subscription %s/%s to slice", requestKey.Namespace, operand.Name)
+	// Create an unstructured object for CR and check its value
+	var crFromRequest unstructured.Unstructured
+
+	if operand.APIVersion == "" {
+		return fmt.Errorf("The APIVersion of operand is empty for operator " + operand.Name)
 	}
-	merr := &util.MultiErr{}
 
-	// Merge OperandConfig and ClusterServiceVersion alm-examples
-	var found bool
-	for _, almExample := range almExampleList {
-		// Create an unstructured object for CR and check its value
-		var crFromALM unstructured.Unstructured
-		crFromALM.Object = almExample.(map[string]interface{})
-		if crFromALM.GetKind() != operand.Kind {
-			continue
-		}
+	if operand.Kind == "" {
+		return fmt.Errorf("The Kind of operand is empty for operator " + operand.Name)
+	}
 
-		found = true
-		var name string
-		if operand.InstanceName == "" {
-			name = requestKey.Name
-		} else {
-			name = operand.InstanceName
-		}
+	var name string
+	if operand.InstanceName == "" {
+		name = requestKey.Name
+	} else {
+		name = operand.InstanceName
```

Comment:
Just wanna to confirm that the CR's name is `instanceName` or name of `OperandRequest`.
The `Spec.Requests[0].Operands[0].Name` is only for operator name.

Is it correct?

Reply:
Yes and no.

`Spec.Requests[0].Operands[0].Name` is not for the operator name.  It is a common name to connect OperandRequest, OperandRegistry and OperandConfig. The operator name is defined in the OperandRegistry.

Reply:
yeah, get it. Others look good to me